python/ray/_private/log_monitor.py

```diff
@@ -76,10 +76,14 @@ def reopen_if_necessary(self):
         open_inode = None
         if self.file_handle and not self.file_handle.closed:
             open_inode = os.fstat(self.file_handle.fileno()).st_ino
-        new_inode = os.stat(self.filename).st_ino
-        if open_inode != new_inode:
-            self.file_handle = open(self.filename, "rb")
-            self.file_handle.seek(self.file_position)
+
+        try:
+            new_inode = os.stat(self.filename).st_ino
+            if open_inode != new_inode:
+                self.file_handle = open(self.filename, "rb")
+                self.file_handle.seek(self.file_position)
+        except Exception:
+            logger.debug(f"file no longer exists, skip re-opening of {self.filename}")
```

Comment:
can you use .info?

Reply:
also we may want to try the whole method since we have `open_inode = os.fstat(self.file_handle.fileno()).st_ino` as well

Reply:
i think info is overkill since it normal that the log file is deleted on rotation

I don't know if there are other side effects if we wrap more code than needed, just wanted to be safe

Reply:
hmm but this will be still broken if `self.file_handle and not self.file_handle.closed:` is executed and file doesn't exist

Reply:
also this particular user who reports the issue doesn't do log rotation. So it is not due to log rotation (it is a bug from other part of code)

Reply:
seems this fix is generally useful, we shouldn't fail if a file is deleted